examples/ticks_and_spines/date_precision_and_epochs.py

```diff
@@ -0,0 +1,114 @@
+"""
+=========================
+Date Precision and Epochs
+=========================
+
+Matplotlib can handle `.datetime` objects and `numpy.datetime64` objects using
+a unit converter that recognizes these dates and converts them to floating
+point numbers. By deafult this conversion returns a float that is days
+since "0000-01-01T00:00:00".  This has resolution implications for modern
+dates:  "2000-01-01" in this time frame is 730120, and a 64-bit floating point
+number has a resolution of 2^{-52}, or approximately 14 microseconds.
+
+"""
+import datetime
+import dateutil
+import numpy as np
+
+import matplotlib
+import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
+
+#############################################################################
+# Datetime
+# --------
+#
+# Python `.datetime` objects have microsecond reesolution, so by default
+# matplotlib dates cannot round-trip full-resolution datetime objects:
+
+date1 = datetime.datetime(2000, 1, 1, 0, 10, 0, 12,
+                          tzinfo=dateutil.tz.gettz('UTC'))
+mdate1 = mdates.date2num(date1)
+print('Before Roundtrip: ', date1, 'Matplotlib date:', mdate1)
+date2 = mdates.num2date(mdate1)
+print('After Roundtrip:  ', date2)
+
+#############################################################################
+# Note this is only a round-off error, and there is no problem for
+# dates closer to the epoch:
+
+date1 = datetime.datetime(10, 1, 1, 0, 10, 0, 12,
+                          tzinfo=dateutil.tz.gettz('UTC'))
+mdate1 = mdates.date2num(date1)
+print('Before Roundtrip: ', date1, 'Matplotlib date:', mdate1)
+date2 = mdates.num2date(mdate1)
+print('After Roundtrip:  ', date2)
+
+#############################################################################
+# If a user wants to use modern dates at micro-second precision, they
+# can change the epoch.
+
+mdates.set_epoch('1990-01-01')
+
+date1 = datetime.datetime(2000, 1, 1, 0, 10, 0, 12,
+                          tzinfo=dateutil.tz.gettz('UTC'))
+mdate1 = mdates.date2num(date1)
+print('Before Roundtrip: ', date1, 'Matplotlib date:', mdate1)
+date2 = mdates.num2date(mdate1)
+print('After Roundtrip:  ', date2)
+
+#############################################################################
+# datetime64
+# ----------
+#
+# `numpy.datetime64` objects have micro-second precision for a much larger
```

Comment:
throughout: pick either "micro-seconds" or "microseconds", but be consistent (Wikipedia says "microseconds").